pkg/util/provider/machinecontroller/machine.go

```
@@ -101,12 +119,18 @@ func (c *controller) reconcileClusterMachineKey(key string) error {
 		return err
 	}
 
+	if machine.DeletionTimestamp != nil {
+		klog.Errorf("should be in machine termination queue %s", machine.Name)
+		return nil
```

Why not return an error here?
A terminating machine in the machineQueue should be eventually removed and added to the proper queue

---

In case the reconcile returns an `err`, the worker/machine is re-queued again for `DefaultMaxRetries` for reconciliation, in this case however since its a machine that shouldn't even be in this queue, further re-queues should be prevented. Hence the error is logged and `nil` is returned.